src/lib/pages/workflow-history-json.svelte

```diff
@@ -1,47 +1,46 @@
 <script lang="ts">
   import { page } from '$app/stores';
-
-  import CodeBlock from '$lib/holocene/code-block.svelte';
-  import Loading from '$lib/holocene/loading.svelte';
   import ToggleSwitch from '$lib/holocene/toggle-switch.svelte';
+
+  import WorkflowJsonNavigator from '$lib/components/workflow/workflow-json-navigator.svelte';
   import { fetchAllEvents, fetchRawEvents } from '$lib/services/events-service';
   import { authUser } from '$lib/stores/auth-user';
   import { decodeURIForSvelte } from '$lib/utilities/encode-uri';
 
   const { namespace, workflow: workflowId, run: runId } = $page.params;
 
   let decodeEventHistory = true;
+  let events = [];
 
-  $: events = decodeEventHistory
-    ? fetchAllEvents({
-        namespace: decodeURIForSvelte(namespace),
-        workflowId: decodeURIForSvelte(workflowId),
-        runId: decodeURIForSvelte(runId),
-        settings: $page.data.settings,
-        accessToken: $authUser?.accessToken,
-        sort: 'ascending',
-      })
-    : fetchRawEvents({
-        namespace: decodeURIForSvelte(namespace),
-        workflowId: decodeURIForSvelte(workflowId),
-        runId: decodeURIForSvelte(runId),
-        sort: 'ascending',
-      });
+  const fetchEvents = async () => {
+    events = decodeEventHistory
+      ? await fetchAllEvents({
+          namespace: decodeURIForSvelte(namespace),
+          workflowId: decodeURIForSvelte(workflowId),
+          runId: decodeURIForSvelte(runId),
+          settings: $page.data.settings,
+          accessToken: $authUser?.accessToken,
+          sort: 'ascending',
+        })
+      : await fetchRawEvents({
+          namespace: decodeURIForSvelte(namespace),
+          workflowId: decodeURIForSvelte(workflowId),
+          runId: decodeURIForSvelte(runId),
+          sort: 'ascending',
+        });
+  };
+  $: decodeEventHistory, fetchEvents();
 </script>
 
-{#await events}
-  <Loading />
-{:then events}
-  <div class="flex justify-end">
-    <label
-      for="decode-event-history"
-      class="flex items-center gap-4 font-secondary text-sm"
-      >View decoded event history<ToggleSwitch
-        id="decode-event-history"
-        bind:checked={decodeEventHistory}
-        data-testid="decode-event-history-toggle"
-      />
-    </label>
-  </div>
-  <CodeBlock content={events} data-testid="event-history-json" />
-{/await}
+<WorkflowJsonNavigator {events}>
+  <label
+    slot="decode"
+    for="decode-event-history"
+    class="flex items-center gap-4 font-secondary text-sm w-96"
```

Comment:
Just some small improvements for responsiveness
```suggestion
    class="flex items-center gap-4 font-secondary text-sm w-96 min-w-fit sm:justify-end"
```

Reply:
Updated in https://github.com/temporalio/ui/pull/1361